piksi_tools/diagnostics.py

```diff
@@ -51,9 +51,12 @@ def __init__(self, link):
                            SBP_MSG_BOOTLOADER_HANDSHAKE_RESP)
     self.link.add_callback(self._print_callback,
                            [SBP_MSG_LOG, SBP_MSG_PRINT_DEP])
+    timeout = time.time() + timeout if timeout is not None else None
     # Wait for the heartbeat
     while not self.heartbeat_received:
       time.sleep(0.1)
+      if timeout is not None and time.time() > timeout:
+        return
```

Comment:
might be useful to print something out and/or exit with -1. Looks good otherwise.